### PR TITLE
Compile field as final if there's a const constructor.

### PR DESCRIPTION
**Pull request by mprobst**

### CODE REVIEW

File: test/DartTest.ts [hunk 1]
```diff
@@ -201,8 +201,11 @@ describe('transpile to dart', () => {
     it('special cases @CONST', () => {
       expectTranslate('@CONST class X {}').to.equal(' const class X { }');
       expectTranslate('@CONST() class X {}').to.equal(' const class X { }');
-      expectTranslate('class X { @CONST constructor() { super(3); this.x = 1; this.y = 2; } }')
-          .to.equal(' class X { const X ( ) : x = 1 , y = 2 , super ( 3 ) ; }');
+      expectTranslate('class X { ' +
```
Comment (alexeagle) [inline test/DartTest.ts]: opt: use backtick multi-line string literal? or is that too hard to indent?

File: test/DartTest.ts [hunk 2]
```diff
@@ -201,8 +201,11 @@ describe('transpile to dart', () => {
...
+                      '  x: number; ' +
+                      '  @CONST constructor() { super(3); this.x = 1; this.y = 2; } ' +
+                      '}')
```
Comment (alexeagle) [inline test/DartTest.ts]: nit: this would be better as a separate assertion with just the member and no statements in the ctor body

**Comment by alexeagle** [comment]
LGTM

File: test/DartTest.ts (hunk 1 above)
Comment (mprobst) [inline test/DartTest.ts]: Done.

File: test/DartTest.ts (hunk 2 above)
Comment (mprobst) [inline test/DartTest.ts]: Well, but the result wouldn't be valid code then (doesn't initialize the final fields). I think it's fine like this.

**Pull request merged**